recipes/argon2/all/conanfile.py

```diff
@@ -0,0 +1,54 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class Argon2Conan(ConanFile):
+    name = "argon2"
+    license = "Apache 2.0", "CC0-1.0"
+    homepage = "https://github.com/P-H-C/phc-winner-argon2"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Argon2 password hashing library"
+    topics = ("argon2", "crypto", "password hashing")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("argon2 recipe currently does not offer Windows support")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("phc-winner-argon2-{0}".format(self.version), "argon2")
+
+    def build(self):
+        arch = str(self.settings.arch).replace("_", "-")
+        with tools.chdir("argon2"):
+            self.run("make OPTTARGET="+arch+" libs")
```

Comment:
Use `self._source_subfolder`.

```suggestion
  with tools.chdir(self._source_subfolder):
    self.run("make OPTTARGET={} libs".format(self._arch))
```